doc/open-extensions/smart-markers.md

```diff
@@ -0,0 +1,39 @@
+When a client enters a conversation after being offline for a while, such client might want to know what was the last message-id that was marked according to the rules defined in [XEP-0333 - Chat Markers][chat-markers], in order to know where he left of, and build an enhanced UI.
+
+MongooseIM provides such functionality, using [mod_smart_markers](../modules/mod_smart_markers.md)
+
+## Namespace
+'esl:xmpp:smart-markers:0'
+
+## Fetching a conversation's latest markers
+
+### Individual fetch
+
+Given a peer, i.e., another user or a muc room, we can fetch the marker we last sent, to the main thread or any other sub-thread, with an IQ like the following:
+```xml
+<iq id='iq-unique-id' type='get'>
+  <query xmlns='esl:xmpp:smart-markers:0' peer='<peer-bare-jid>' [thread='<thread-id>' after='<RFC3339-timestamp>'] />
+</iq>
+```
+where:
+
+* `<peer-bare-jid>` MUST be the bare jid of the peer whose last marker wants to be checked.
+* `<thread>` is an optional attribute that indicates if the check refers to specific a thread in the conversation. If not provided, defaults to the main conversation thread.
+* `<after>` is an optional attribute indicating whether markers sent only after a certain timestamp are desired.
```

Comment:
Please add some note that this field makes sense for group chats only.

Reply:
Wait, why only for groupchats? I might want to know which one was the last marker I sent to a regular user, just as much as to a room as well, and filtering by thread or timestamp applies the same 🤔

Reply:
just add one _actual_ example for each possible destination. I.e. for groupchat and for one2one. Including a date as a date ;) 

Something like:

```
# Query for markers in a room

Request
Result

# Query for markers in one2one conversation

Request
Result
```

Reply:
It can be a generic implementation, but it brings the most value for group chats. Since there will be not more than just 2 chat markers for 1-2-1 chats.